src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/BackupRestoreItem.java

```diff
@@ -164,6 +181,28 @@ public void retrieveInterstepData(StepExecution stepExecution) {
 
         JobParameters jobParameters = this.currentJobExecution.getJobParameters();
 
+        boolean parameterizePasswords = Boolean.parseBoolean(
+            jobParameters.getString(Backup.PARAM_PARAMETERIZE_PASSWDS, "false"));
+
+
+        if (parameterizePasswords) {
+
+            //here we set some customized XML handling code. For backups, we add a converter that tokenizes
+            //outgoing passwords. for restores, a handler for those tokenized backups.
```

Comment:
Given that one of the converters only marshals, and the other only unmarshals, you could make an argument for combining them, but they are probably fine as-is.

Reply:
Unfortunately the marshalling/unmarshalling are mutually exclusive, otherwise I'd combine them.